src/tests.rs

```diff
@@ -16,15 +19,24 @@ fn spot_price_should_work() {
     }
 }
 
+
 #[test]
 fn sell_price_should_work() {
     let cases = vec![
         (1000, 2000, 500, Some(667), "Easy case"),
-        (0, 0, 0, None, "Zero reserves and weights"),
-        (0, 1, 0, None, "Zero sell reserve and amount"),
-        (1, 0, 0, Some(1), "Zero buy reserve and amount"),
-        (0, 0, u128::MAX, Some(1), "Zero buy reserve and sell reserve"),
         (0, u128::MAX, u128::MAX, None, "Zero sell reserve"),
+
```

Comment:
for #calculate_sell_price the formula is:
/// Formula : BUY_RESERVE * SELL_AMOUNT / (SELL_RESERVE + SELL_AMOUNT )

So if either buy_reserve or sell_amount is zero, the denominator will be zero and again we dont have to compute anyhting. 
Now if we include a conditional for these cases, the behaviour of our method changes quite drastically: